models/evc.py

```diff
@@ -825,7 +830,17 @@ def should_deploy(self, path=None):
 
         return False
 
-    def deploy_to_path(self, path=None):  # pylint: disable=too-many-branches
+    def error_hapenned(self):
```

Comment:
Can this method be named more closely for what it does? By just reading `error_happend()` at call sites you have no idea what will happen. Hindering code readability. Maybe something like `disable_set_error(error: str)` -> None? (assuming the error str discussion)

Reply:
Also, thinking more about it generally, `self.disable()` should really only be used when a user needs to admin disable it (with an update operation) or when a deletion via request happen. I think we should maintain the semantics of `disabled` as it always been. 

All the other retriable failable cases, it should only deactivate, and then set the error accordingly that can still recover it from. Basically the recover for any link_down|link_up is always a redeploy to be fully sure. For a request deletion that only delete from one switch but failed on the other switches should still only deactivate too, and then set the reason with something related that it needs to be deleted again (this last part here can get a bit better if we we're to support in a single request a list of dpid as you suggested before, but considering we're wrapping up the release it'll probably not be doable to fully cover that part but we can plan for it incrementally), either way, using a correct status_reason here too I can see it working.

Final piece of advice: I highly recommend you thinking through the cases again, and organize how we'll use the status_reason values. Happy to keep providing these partial reviews to help the PR to land, but for these cases it's recommended we sort it out first to avoid much back and forth and reimplementations.

Reply:
Name changed with `deactivate_set_error()`. It no longer disables the EVC. So far, I have no noticed any problem with kytos redeploying for motived like `link_(down|up)` for an EVC with errors.